src/test/scala/org/broadinstitute/dsde/workbench/sam/service/StatusServiceSpec.scala

```diff
@@ -25,52 +28,74 @@ class StatusServiceSpec extends AnyFreeSpec with Matchers with BeforeAndAfterAll
   val allUsersEmail = WorkbenchEmail("allusers@example.com")
   val dbReference = TestSupport.dbRef
 
+  val directoryConfig: DirectoryConfig = TestSupport.directoryConfig
+  val dirURI = new URI(directoryConfig.directoryUrl)
+  val connectionPool = new LDAPConnectionPool(new LDAPConnection(dirURI.getHost, dirURI.getPort, directoryConfig.user, directoryConfig.password), directoryConfig.connectionPoolSize)
+
   override def afterAll(): Unit = {
+    connectionPool.close()
     system.terminate()
   }
 
-  private def newStatusService(directoryDAO: DirectoryDAO) = {
-    new StatusService(directoryDAO, new NoExtensions {
+  // We need to mock this because Postgres does some automatic setup and gets unhappy with multiple instances of its DAO existing at once.
+  private def mockDirectoryDAOWithPostgresStatusCheck(dbReferenceOverride: DbReference = dbReference) = {
```

Comment:
Same comment here re: duplicating the `checkStatus` implementation

Reply:
So, this one we can't use the real postgres db, because postgres makes some assumptions that it will only get stood up once within the whole test suite, and we don't want to use a persistent postgres across all the suites here, because we want to have different DBRef objects to manage bringing it down or up, so we can't actually override directoryDao itself. (or at least, I couldn't figure out how to)

Reply:
Could we fully mock out the DAOs here then? It doesn't feel right to be overriding our mocks with the real implementation. I'd prefer that we either use the real implementation with a real `PostgresDirectoryDAO` or we use the mock rather than relying on this combination. This goes for the `MockDirectoryDAO` in `TestSamRoutes` as well